### PR TITLE
[Bug] Habit isn't automatically transferred to the section of 'Acquired habits' if it has 100% progress /bug#2243

### PR DESCRIPTION
**Pull request by yuriiLisunov**
**Description:**

**BEFORE:**
1.Habit isn't automatically transferred to the section of 'Acquired habits' from 'Habits in progress' if it has 100% progress
2."Habit_streak" and 'Duration' is represented on card of 'Habits in progress'.
3. 'Mark as done' switcher is is represented on card of 'Habits in progress'
4. 'Mark as done' label is s represented on card of 'Habits in progress'

**AFFTER:**
1.Habbit is automatically transferred to the section of 'Acquired habits', and not represented in 'Habits in progress'.
Counter of 'Habits progress': -1.Counter of 'Acquired habits': +1
2.'Duration' is represented on card of 'Acquired habits'.
3.Star' icon is represented on card of 'Acquired habits'.
4."Great! You've got a habit!" text is represented on card of 'Acquired habits'.

acquired

### CODE REVIEW

File: src/app/component/user/components/profile/profile-dashboard/one-habit/one-habit.component.ts [hunk 1]
```diff
@@ -76,11 +78,16 @@ export class OneHabitComponent implements OnInit {
...
       .subscribe(response => {
+        if (response.status === 'ACQUIRED') {
+          this.descriptionType.acquired();
+          this.nowAcquiredHabit.emit(response);
+        } else {
         this.habit.habitStatusCalendarDtoList = response.habitStatusCalendarDtoList;
         this.habit.workingDays = response.workingDays;
         this.habit.habitStreak = response.habitStreak;
         this.buildHabitDescription();
         this.isRequest = false;
```
Comment (JimHolman1199) [inline src/app/component/user/components/profile/profile-dashboard/one-habit/one-habit.component.ts:89]: add correct formating to this part

File: src/app/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts [hunk 2]
```diff
@@ -34,6 +34,12 @@ export class ProfileDashboardComponent implements OnInit, OnDestroy {
     this.getUserId();
   }
 
+  public changeStatus(response) {
+    const index = this.habitsInProgress.findIndex((habitInProg) => habitInProg.id === response.id);
+    this.habitsInProgress.splice(index, 1);
+    this.habitsAcquired.unshift(response);
```
Comment (JimHolman1199) [inline src/app/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts:40]: use arr.filter and destructuring here

Comment (yuriiLisunov) [reply src/app/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts]: fixed

File: src/app/component/user/components/profile/profile-dashboard/one-habit/one-habit.component.ts (hunk 1 above)
Comment (yuriiLisunov) [reply src/app/component/user/components/profile/profile-dashboard/one-habit/one-habit.component.ts]: added

File: src/app/component/user/components/profile/profile-dashboard/one-habit/one-habit.component.ts [hunk 3]
```diff
@@ -74,14 +76,19 @@ export class OneHabitComponent implements OnInit {
...
-      .subscribe(response => {
-        this.habit.habitStatusCalendarDtoList = response.habitStatusCalendarDtoList;
-        this.habit.workingDays = response.workingDays;
-        this.habit.habitStreak = response.habitStreak;
-        this.buildHabitDescription();
-        this.isRequest = false;
-      });
+    .pipe(take(1))
+    .subscribe(response => {
+      if (response.status === 'ACQUIRED') {
```
Comment (JimHolman1199) [inline src/app/component/user/components/profile/profile-dashboard/one-habit/one-habit.component.ts:81]: ACQUIRED should be in enum file.
so find one or create

File: src/app/model/habit/HabitStatus.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export enum HabitStatus {
```
Comment (JimHolman1199) [inline src/app/model/habit/HabitStatus.ts:1]: Hi, could you please rename file from HabitStatus.ts to HabitStatus.enum.ts

**Review by idzhersh — APPROVED** [review]

**Review by JimHolman1199 — APPROVED** [review]

**Review by VitalyBorysenko — APPROVED** [review]

**Review by JimHolman1199 — APPROVED** [review]

**Review by paslavskyi9roman — APPROVED** [review]

**Review by KirillAbramo — APPROVED** [review]

**Review by mr-pavlenko — APPROVED** [review]

**Review by AlekOK — COMMENTED** [review]
ok

**Pull request merged**